lib/hybrid.h

```diff
@@ -419,7 +433,12 @@ struct HybridSimulator final {
       const auto& gatex = gatexs[i];
       unsigned k = (mask >> shift_length) & ((1 << gatex.schmidt_bits) - 1);
       shift_length += gatex.schmidt_bits;
+
       if (k != prev_k[i]) {
+        if (k >= gatex.schmidt_decomp.size()) {
+          return false;
```

Comment:
From what I can tell, if this `return false` is hit then the system is in an error state, since it prevents any subsequent gates in `gatexs` from being used to update `prev_k`. For example:
```
Let gatexs.size() == 10.
Assume that this check fails at i == 2 and returns.
Then gatexs[3] to gatexs[9] are not checked.
Thus, prev_k[3] to prev_k[9] will not be correct.
```
Is this reading correct? How does this interact with the `continue`s above?

Reply:
No, this reading is incorrect. We return false for an invalid path, see above. In this case, prev_k[3] to prev_k[9] will not be affected at all as we increment `mask` by 1 (`mask` encodes the path) and the path bits for gatexs[3] to gatexs[9] remain the same when the invalid path event is triggered for `i == 2`. There is a possibility to speed up things here by not iterating over the bits for `i < 2`. I will do this in some later PR as now we don't have gates with Schimdt rank 3 and it will not affect anything.

Reply:
Sounds good to me.